jax_galsim/position.py

```diff
@@ -0,0 +1,150 @@
+import jax.numpy as jnp
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+
+@_wraps(_galsim.Position)
+class Position(object):
+    def __init__(self, *args, **kwargs):
```

Comment:
GalSim does not allow to instantiate the base class. I think it is because you either use `PositionD` or `PositionI`. Maybe it can cause issues if some part of the library is expecting either `PositionD` or `PositionI` and not `Position`

Reply:
Related question: since we allow to use `Position`, why do we use `@register_pytree_node_class` only for `PositionD` and `PositionI`?

Reply:
You are right, the reason I'm doing it like this is because of the mechanism to flatten and unflatten the data structure that forces me to use the init function, it cannot be just a NotImplemented trigger